src/main/java/seedu/clinic/model/CommandHistoryList.java

```diff
@@ -0,0 +1,87 @@
+package seedu.clinic.model;
+
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.clinic.storage.TextFileCommandHistoryStorage;
+
+public class CommandHistoryList {
+    private final ObservableList<String> internalList = FXCollections.observableArrayList();
+    private int commandHistoryIndex = internalList.size();
```

Comment:
Just some notes: I guess the commandHistoryIndex can be of type Index in the future, but looking at the Index class it does not have increment/decrement, so currently it looks fine to just use the int type here.